chatmaild/src/chatmaild/rm_accounts.py

```diff
@@ -0,0 +1,54 @@
+import crypt
+import json
+import logging
+import os
+import sys
+import time
+import shutil
+from pathlib import Path
+from socketserver import (
+    StreamRequestHandler,
+    ThreadingMixIn,
+    UnixStreamServer,
+)
+
+from .config import Config, read_config
+from .database import Database
+
+
+def remove_users(db: Database, cutoff_date: int):
+    db.connect()
+    try:
+        delete_query = "DELETE FROM users WHERE last_login <?"
+        db.execute_query(delete_query, (cutoff_date))
+        db.commit_changes()
+    finally:
+        db.close()
+
+
+def remove_user_data(db: Database, cutoff_date: int, dir: Path):
+    """Collects all users where last_login < cutoff_date and deletes corresponding directories."""
+    db.connect()
+
+    try:
+        select_query = "SELECT user FROM users WHERE last_login <?"
+        cursor = db.execute_query(select_query, (cutoff_date,))
+        usernames = cursor.fetchall()
+
+        for username in usernames:
+            user_dir = dir / username[0]
```

Comment:
don't use `dir` -- it's a builtin name and you are shadowing it. 
Also `vmail_basedir` would be more telling.